### DISCUSSION TITLE
The kernel appears to have died. It will restart automatically. HGTLoader !

### DISCUSSION THREAD

**Discussion by dinhhoi** [pyg-team/pytorch_geometric#5178] [Q&A]
Hello, 
I am trying to create my own heterogeneous graph. 
I successfully replicate the OGG_MAG example. but have trouble running my own datasets

train_loader = HGTLoader(data, num_samples=[1024] * 4, shuffle=False,
                             input_nodes=train_input_nodes, **kwargs)
Whenever I tried to run the HGTLoader, "the kernel appears to have died. It will restart automatically" 
m

My torch version, cuda, torchvision, torchaudio, 
1.11.0+cu113
11.3
0.12.0+cu113
0.11.0+cu113

I used pip to install packages: 

![image](https://user-images.githubusercontent.com/34597401/183696402-ceed2e65-bb9f-466a-ab41-1e9172cc9af2.png)

!pip install torch==1.11.0+cu113 torchvision==0.12.0+cu113 torchaudio==0.11.0 --extra-index-url https://download.pytorch.org/whl/cu113

!pip install torch-scatter -f https://data.pyg.org/whl/torch-1.11.0+cu113.html
!pip install torch-sparse -f https://data.pyg.org/whl/torch-1.11.0+cu113.html
!pip install torch-geometric
!pip install torch-cluster -f https://data.pyg.org/whl/torch-1.11.0+cu113.html
!pip install torch-spline-conv -f https://data.pyg.org/whl/torch-1.11.0+cu113.html

Also, the terminal just showed:"[IPKernelApp] WARNING | Unknown error in handling startup files:" so I opened `ipython` terminal to check the error, but nothing is shown there. 

Please let me know how I can fix my issue.

**Comment by rusty1s** [comment]
Thanks for reporting. Does swapping out `HGTLoader` with `NeighborLoader` resolves this error? Trying to figure out if this issue is related to the loader or the data.

**Reply by dinhhoi** [reply]
No, I already tried the NeighborLoader. still the same issue. 
I have followed the docs to make sure my own dataset is constructed correctly, but not 100% confident. Is there any document/source you suggest? appreciate your help. 

Thanks

**Reply by rusty1s** [reply]
Can you share your `data` with me? Can you also confirm that your data is formatted correctly? The following should run through:
```python
for edge_type in data.edge_types:
    src, _, dst = edge_type
    assert data[edge_type].edge_index[0].max() < data[src].num_nodes
    assert data[edge_type].edge_index[1].max() < data[dst].num_nodes

**Reply by dinhhoi** [reply]
Hi Rusty1s, my issue has been resolved. I tried creating my own tabular data. So I have to reset the index and the remap to have the correct format of data. 
That would be great if there is one official instruction in the documents. 

Thanks,
